lib/runner/extensions/item.command.js

```diff
@@ -88,6 +97,13 @@ module.exports = {
                         });
                     }
 
+                    // update allowed request mutation properties with the mutated context
+                    // @note from this point forward, make sure this mutated
+                    // request instance is used for upcoming commands.
+                    ALLOWED_REQUEST_MUTATIONS.forEach(function (property) {
+                        item.request[property] = ctxTemplate.request[property];
```

Comment:
HeaderList has parent reference. You’d need to account for it. Check if there is any helper method to move lists.